app/serialization.py

```diff
@@ -40,9 +41,9 @@
 )
 
 
-def deserialize_host(raw_data):
+def deserialize_host(raw_data, schema):
     try:
-        validated_data = HostSchema(strict=True).load(raw_data).data
+        validated_data = schema.load(raw_data).data
```

Comment:
It’s this function’s responsibility to validate the schema in a way that it raises a _ValidationException_. Thus a Schema class should be passed in instead of its instance.

```suggestion
        validated_data = schema(strict=True).load(raw_data).data
```